Facing "panic: assignment to entry in nil map" when trying to export otlp logs to google cloud

i have the following configuration for my otel collector 
```yaml
extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
    endpoint: 0.0.0.0:55679

receivers:
  otlp:
    protocols:
      grpc:
      http:
        endpoint: 0.0.0.0:4318
        cors:
          allowed_origins:
            - "*"

processors:
  batch:

exporters:
  logging:
    loglevel: debug
  googlecloud:
    retry_on_failure:
      enabled: false
    project: project-name
    log:
      default_log_name: opentelemetry.io/collector-exported-log


service:
  telemetry:
    logs:
      level: debug
  pipelines:
    logs:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, googlecloud]

  extensions: [health_check, pprof, zpages]

```
When I run the collector with just the loggin exporter in my logs pipeline, things work fine and i can see the logs in the otel stdout, but when i run it with googlecloud exporter, i get the following error 

```go
panic: assignment to entry in nil map

goroutine 145 [running]:
github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector.logMapper.logToSplitEntries({{0xc000682420}, {{{0xc00029a2a0, 0x27}, {0x0, {...}, 0x0}}, {0xc0009c9818, 0x11}, {0x4c4bb10, 0x2b}, ...}, ...}, ...)
	github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector@v0.30.2-0.20220512190557-875976f87023/logs.go:288 +0xd7
github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector.logMapper.createEntries({{0xc000682420}, {{{0xc00029a2a0, 0x27}, {0x0, {...}, 0x0}}, {0xc0009c9818, 0x11}, {0x4c4bb10, 0x2b}, ...}, ...}, ...)
	github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector@v0.30.2-0.20220512190557-875976f87023/logs.go:201 +0x431
github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector.(*LogsExporter).PushLogs(0xc000878900, {0x5492a30, 0xc000c42210}, {0x0?})
	github.com/GoogleCloudPlatform/opentelemetry-operations-go/exporter/collector@v0.30.2-0.20220512190557-875976f87023/logs.go:133 +0xbb
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsRequest).export(0x7fe726bbb8e0?, {0x5492a30?, 0xc000c42210?})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/logs.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc000c84a50, {0x54a5d40, 0xc000c421b0})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/common.go:225 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc0003f3200, {0x54a5d40?, 0xc000c421b0?})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/queued_retry.go:147 +0x156
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsExporterWithObservability).send(0xc000910498, {0x54a5d40, 0xc000c421b0})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/logs.go:131 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x472ba20?, 0xc000c421b0})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/queued_retry_inmemory.go:119 +0x5c
go.opentelemetry.io/collector/exporter/exporterhelper/internal.consumerFunc.consume(0x0?, {0x472ba20?, 0xc000c421b0?})
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/internal/bounded_memory_queue.go:82 +0x2c
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func2()
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/internal/bounded_memory_queue.go:69 +0xcd
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
	go.opentelemetry.io/collector@v0.53.0/exporter/exporterhelper/internal/bounded_memory_queue.go:63 +0x98
```

I'm facing this same error in local as well as production environments.

as far as i can tell this error only occurs when using the google cloud exporter for exporting logs. Exporting traces seems to be working fine.

--- a/exporter/collector/integrationtest/testcases.go
+++ b/exporter/collector/integrationtest/testcases.go
@@ -36,6 +36,11 @@ var (
 			OTLPInputFixturePath: "testdata/fixtures/logs_multi_project.json",
 			ExpectFixturePath:    "testdata/fixtures/logs_multi_project_expected.json",
 		},
+		{
+			Name:                 "Logs with scope information",
+			OTLPInputFixturePath: "testdata/fixtures/logs_apache_error_scope.json",
+			ExpectFixturePath:    "testdata/fixtures/logs_apache_error_scope_expected.json",
+		},
 	}
 
 	MetricsTestCases = []TestCase{

--- a/exporter/collector/integrationtest/testdata/fixtures/logs_apache_error_scope.json
+++ b/exporter/collector/integrationtest/testdata/fixtures/logs_apache_error_scope.json
@@ -1,0 +1,69 @@
+{
+  "resourceLogs": [
+    {
+      "resource": {
+        "attributes": [
+          {
+            "key": "cloud.platform",
+            "value": {
+              "stringValue": "gcp_compute_engine"
+            }
+          }
+        ]
+      },
+      "scopeLogs": [
+        {
+          "scope": {
+            "name": "scopeNameFoo",
+            "version": "9000"
+          },
+          "logRecords": [
+            {
+              "timeUnixNano": "1651013315606298000",
+              "body": {
+                "kvlistValue": {
+                  "values": [
+                    {
+                      "key": "time",
+                      "value": {
+                        "stringValue": "Tue Apr 26 22:48:35.606298 2022"
+                      }
+                    },
+                    {
+                      "key": "severity",
+                      "value": {
+                        "stringValue": "notice"
+                      }
+                    },
+                    {
+                      "key": "message",
+                      "value": {
+                        "stringValue": "[pid 769:tid 140159306111872] AH00094: Command line: '/usr/local/apache/bin/httpd'"
+                      }
+                    }
+                  ]
+                }
+              },
+              "attributes": [
+                {
+                  "key": "log.file.name",
+                  "value": {
+                    "stringValue": "test.log"
+                  }
+                },
+                {
+                  "key": "gcp.log_name",
+                  "value": {
+                    "stringValue": "apache-error-fixture"
+                  }
+                }
+              ],
+              "traceId": "",
+              "spanId": ""
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}

--- a/exporter/collector/integrationtest/testdata/fixtures/logs_apache_error_scope_expected.json
+++ b/exporter/collector/integrationtest/testdata/fixtures/logs_apache_error_scope_expected.json
@@ -1,0 +1,30 @@
+{
+  "writeLogEntriesRequests": [
+    {
+      "entries": [
+        {
+          "logName": "projects/fakeprojectid/logs/apache-error-fixture",
+          "resource": {
+            "type": "gce_instance",
+            "labels": {
+              "instance_id": "",
+              "zone": ""
+            }
+          },
+          "jsonPayload": {
+            "message": "[pid 769:tid 140159306111872] AH00094: Command line: '/usr/local/apache/bin/httpd'",
+            "severity": "notice",
+            "time": "Tue Apr 26 22:48:35.606298 2022"
+          },
+          "timestamp": "1970-01-01T00:00:00Z",
+          "labels": {
+            "instrumentation_source": "scopeNameFoo",
+            "instrumentation_version": "9000",
+            "log.file.name": "test.log"
+          }
+        }
+      ],
+      "partialSuccess": true
+    }
+  ]
+}

--- a/exporter/collector/logs.go
+++ b/exporter/collector/logs.go
@@ -295,14 +295,6 @@ func (l logMapper) logToSplitEntries(
 		Resource: mr,
 	}
 
-	// TODO(damemi): Make overwriting these labels (if they already exist) configurable
-	if len(instrumentationSource) > 0 {
-		entry.Labels["instrumentation_source"] = instrumentationSource
-	}
-	if len(instrumentationVersion) > 0 {
-		entry.Labels["instrumentation_version"] = instrumentationVersion
-	}
-
 	// if timestamp has not been explicitly initialized, default to current time
 	// TODO: figure out how to fall back to observed_time_unix_nano as recommended
 	//   (see https://github.com/open-telemetry/opentelemetry-proto/blob/4abbb78/opentelemetry/proto/logs/v1/logs.proto#L176-L179)
@@ -352,6 +344,20 @@ func (l logMapper) logToSplitEntries(
 		return []logging.Entry{entry}, fmt.Errorf("Unknown SeverityNumber %v", log.SeverityNumber())
 	}
 	entry.Severity = severityMapping[log.SeverityNumber()]
+
+	if entry.Labels == nil &&
+		(len(instrumentationSource) > 0 ||
+			len(instrumentationVersion) > 0) {
+		entry.Labels = make(map[string]string)
+	}
+
+	// TODO(damemi): Make overwriting these labels (if they already exist) configurable
+	if len(instrumentationSource) > 0 {
+		entry.Labels["instrumentation_source"] = instrumentationSource
+	}
+	if len(instrumentationVersion) > 0 {
+		entry.Labels["instrumentation_version"] = instrumentationVersion
+	}
 
 	// parse remaining OTel attributes to GCP labels
 	for k, v := range attrsMap {

